torch/csrc/jit/custom_operator.h

```diff
@@ -0,0 +1,190 @@
+#pragma once
+
+#include <torch/csrc/jit/function_schema.h>
+#include <torch/csrc/jit/ir.h>
+#include <torch/csrc/jit/operator.h>
+#include <torch/csrc/jit/stack.h>
+#include <torch/csrc/utils/variadic.h>
+
+#include <caffe2/utils/Metaprogramming.h>
+#include <caffe2/utils/TypeList.h>
+
+namespace torch { namespace jit {
+namespace detail {
+template <typename... Ts, size_t... Is>
+std::vector<Argument> createArgumentVectorFromTypes(Indices<Is...> indices) {
```

Comment:
It might make sense to generalize TypeList::map_types_to_values so it can return a custom type, then just tell it return `vector` instead of `tuple` and you can skip all custom metaprogramming here.